### ISSUE TITLE
Precedence bug

### ISSUE THREAD

**Issue by eserte** [regru/Net-Whois-Raw#17]
The change 16866130efc93d066114e0c66b972fae16816308 introduced a precedence bug. Writing
```
 elsif ( my ( $rs ) = _referral_server() && !$norecurse ) {
```
means the same as
```
 elsif ( my ( $rs ) = (_referral_server() && !$norecurse) ) {
```
which is very probably wrong (and breaks one use case here with QRY_ALL).

**Comment by eserte** [comment]
Possible fix (sorry, no formal PR for some dubious reasons):
```
From fec0bf9ddd66cfc5dc4a050f018519b812ebd7bf Mon Sep 17 00:00:00 2001
From: Slaven Rezic <srezic@cpan.org>
Date: Fri, 21 Sep 2018 15:54:21 +0200
Subject: [PATCH] fix for precedence issue (github #17)

---
 lib/Net/Whois/Raw.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Net/Whois/Raw.pm b/lib/Net/Whois/Raw.pm
index b2b6d72..4946964 100644
--- a/lib/Net/Whois/Raw.pm
+++ b/lib/Net/Whois/Raw.pm
@@ -182,7 +182,7 @@ sub recursive_whois {
         elsif ( $whois =~ /To single out one record, look it up with \"xxx\",/s ) {
             return recursive_whois( "=$dom", $srv, $was_srv );
         }
-        elsif ( my ( $rs ) = _referral_server() && !$norecurse ) {
+        elsif ( (my ( $rs ) = _referral_server()) && !$norecurse ) {
             $newsrv = $rs;
             last;
         }
-- 
2.7.4

```

**Comment by nalobin** [comment]
Hello and thanx for report! Net-Whois-Raw-2.99019 fix this issue.
